### PR TITLE
Added Mac-specific text.

### PR DESCRIPTION
**Pull request by cxbrooks**
I updated boards/hail/README.md with what was necessary for me to get the hail board to work under Mac OS X.

Feel free to deny this pull request if it does not meet the style of the README.md files.

### CODE REVIEW

File: boards/hail/README.md [hunk 1]
```diff
@@ -40,3 +111,26 @@ You can also specify the serial port to use:
 ```bash
 make program PORT=/dev/ttyUSB0
 ```
+
+Under Mac OS X, plug in the board and use `ls -ltr /dev/tty*` to find a recently created device:
```
Comment (alevy) [inline boards/hail/README.md]: I think @cxbrooks may have been shipped a hail board that had not been properly provisioned (as was I). A fully provisioned hail should have the FTDI firmware modified to have a product name which includes the string "Tock" and `tockloader` (which was written for hail) looks for such a device and chooses it by default.

So on a properly provisioned hail you shouldn't need any of this section. Is that right @bradjc @brghena?

**Comment by alevy** [comment]
This is great! Thanks! I just want to wait for a tiny bit of feedback from the hail folks to see if this is all accurate or just a bug in the provisioning of the hail you were shipped (as per my inline comment).

Might be useful to move some of this to a higher-level README as some of it is not hail-specifc, but good here for now.

File: boards/hail/README.md (hunk 1 above)
Comment (cxbrooks) [inline boards/hail/README.md]: The hail boards I received have MAC addresses on the back and were running the blink program.  I believe that it was unnecessary for me to run `make program`.

Probably the text should be updated to state something about this.  Something like:

`The hail board are typically shipped provisioned and with the blink userland program running, so it is not typically necessary to rerun `make program`.  However, if your hail board is not provisioned, or you want to reinstall the kernel, then see the steps below.`

The text could be moved to a higher level file.  I originally was starting with boards/hail/README_MAC.md, but decided that one file was better than two.  I was trying to write it to be useful for the Mac user who just received a Hail and was trying to get started.

**Comment by bradjc** [comment]
@brghena reminded me that the feature of Tockloader that checks for a Tock compatible board is in version 0.2.0, not 0.1.0 (which was in pypi). If you run:

    sudo pip-3.6 install tockloader --upgrade

you should get version 0.2.0 which will hopefully search for the port correctly.

**Comment by brghena** [comment]
@cxbrooks having tried this on a macbook today, having the correct Tockloader version should definitely solve the selecting device issue. And I think in general macports is going to be harder to get working than brew.

I'm going to keep some of the text here included in the Makefile and move some of it to an issue so that it can be searchable by someone having problems.

**Pull request closed without merging**